test/core/tsi/s2a/s2a_record_protocol_test.cc

```diff
@@ -245,12 +246,268 @@ static void test_create_crypter_success(TLSCiphersuite ciphersuite) {
   grpc_core::Delete<grpc_channel>(channel);
 }
 
+static void test_encrypt_record_bad_size(TLSCiphersuite ciphersuite) {
+  s2a_crypter* crypter = nullptr;
+  grpc_channel* channel = grpc_core::New<grpc_channel>();
+  char* error_details = nullptr;
+  grpc_status_code status =
+      setup_crypter(ciphersuite, channel, &crypter, &error_details);
+  if (ciphersuite == TLS_CHACHA20_POLY1305_SHA256_ciphersuite) {
+    GPR_ASSERT(status == GRPC_STATUS_UNIMPLEMENTED);
+    int correct_error_message = strcmp(
+        error_details, "The CHACHA-POLY AEAD crypter is not yet implemented.");
+    GPR_ASSERT(correct_error_message == 0);
+
+    // Cleanup.
+    s2a_crypter_destroy(crypter);
+    gpr_free(error_details);
+    grpc_core::Delete<grpc_channel>(channel);
+    return;
+  }
+  GPR_ASSERT(status == GRPC_STATUS_OK);
```

Comment:
Unfortunately I can't move these 2 asserts before the if statement. The reason is that, if the CHACHA-POLY ciphersuite is used, then e.g. `status` will not be `GRPC_STATUS_OK`, rather it will be `GRPC_STATUS_UNIMPLEMENTED`. I don't want the test to fail in the case that the CHACHA-POLY ciphersuite is used.